src/Database/Database.php

```diff
@@ -2698,15 +2698,18 @@ public function getDocument(string $collection, string $id, array $queries = [])
 
         $validator = new Authorization(self::PERMISSION_READ);
         $documentSecurity = $collection->getAttribute('documentSecurity', false);
-        $cacheKey = $this->cacheName . '-cache-' . $this->getNamespace() . ':' . $this->adapter->getTenant() . ':' . $collection->getId() . ':' . $id;
 
+        /**
+         * Cache hash keys
+         */
+        $documentCacheBaseKey = $this->cacheName . '-cache-' . $this->getNamespace() . ':' . $this->adapter->getTenant() . ':' . $collection->getId();
+        $documentCacheKey = $documentCacheHash = $documentCacheBaseKey . ':' . $id;
+        $collectionCacheKey = $documentCacheBaseKey . ':collection';
```

Comment:
```suggestion
        $collectionCacheKey = $documentCacheBaseKey;
```
do we need to append `:collection` here? Can we change `$documentCacheBaseKey` to `$documentCacheBaseKey = $this->cacheName . '-cache-' . $this->getNamespace() . ':' . $this->adapter->getTenant() . ':collection:' . $collection->getId()`